pkg/api/defaults-apiserver.go

```diff
@@ -37,6 +36,11 @@ func (cs *ContainerService) setAPIServerConfig() {
 		"--enable-bootstrap-token-auth": "true",
 		"--v":                           "4",
 	}
+	// if using local etcd server then we need the ca file
+	/*this ugly if statement is made this way, because this function is used in a test that does not pass correct data structure */
```

Comment:
which test is it? Can we fix it?

Reply:
The test strategy needs to be revised. there are a lot of nulls passed. I am not sure if we want to include this `fix` as part of this PR.

Reply:
For sure, not asking that we fix it but I was curious to know which test required this. Could you please open an issue so we don't forget?